WcaOnRails/app/models/competition.rb

```diff
@@ -242,9 +242,12 @@ def persisted_events_id
   # Only validate on update: nobody can confirm competition on creation.
   # The only exception to this is within tests, in which case we actually don't want to run this validation.
   validate :schedule_must_match_rounds, if: :confirmed_at_changed?, on: :update
+  # Competitions after 2018-7-24 will have this check, date set to avoid errors at old competitions
```

Comment:
Please add a comment that explains why you chose a specifc date in the middle of 2018 when this additional code gets added in 2021.

If there is no short'n'sweet explanation, at least add a comment along the lines of "refer to mail thread XY" or "based on a decision by WCAT" or something.

Reply:
I changed this now to be 2018-12-31 as this is where all competitions required a schedule and added a reference why this is now needed per the email threads.